linalg/cudss.cpp

```diff
@@ -0,0 +1,372 @@
+#include "cudss.hpp"
+#include "../general/communication.hpp"
+#include <string>
+
+#ifdef MFEM_USE_CUDSS
+#ifdef MFEM_USE_MPI
+
+namespace mfem
+{
+namespace
+{
+void CheckCudssAPI(cudssStatus_t status, std::string msg)
+{
+   MFEM_VERIFY(status == CUDSS_STATUS_SUCCESS,
+               "mpi rank [" << Mpi::WorldRank() << "] "
+               << "CUDSS call ended unsuccessfully with status = "
+               << static_cast<int>(status)
+               << ", details : " << msg << " Failed!");
+#ifdef MFEM_DEBUG
+   mfem::out << "mpi rank [" << Mpi::WorldRank() << "] "
+             << "CUDSS call ended successfully! "
+             << "Details : " << msg << " successfully!\n";
+#endif
+}
+
+void CheckCudaAPI(cudaError_t status, std::string msg)
+{
+   MFEM_VERIFY(status == cudaSuccess,
+               "mpi rank [" << Mpi::WorldRank() << "] "
+               << "CUDA call ended unsuccessfully with status = "
+               << static_cast<int>(status)
+               << ", details : " << msg << " Failed!");
+#ifdef MFEM_DEBUG
+   mfem::out << "mpi rank [" << Mpi::WorldRank() << "] "
+             << "CUDA call ended successfully! "
+             << "Details : " << msg << " successfully!\n";
+#endif
+}
+} // namespace
+
+CuDSSSolver::CuDSSSolver(MPI_Comm comm_)
+{
+   Init(comm_);
+   InitHandle();
+}
+
+void CuDSSSolver::Init(MPI_Comm comm_)
+{
+   mpi_comm = (MPI_Comm *)malloc(sizeof(MPI_Comm));
+   mpi_comm[0] = comm_; // MPI_COMM_WORLD
+   MPI_Comm_rank(comm_, &myid);
```

Comment:
`myid` does not appear to be needed; no reason to have `mpi_comm` as a pointer. Can remove `CuDSSSolver::Init` entirely, and just initialize `mpi_comm` in the constructor